src/main/java/org/sagebionetworks/bridge/models/assessments/AssessmentReference.java

```diff
@@ -13,20 +14,24 @@
     
     private final ConfigResolver resolver;
     private final String guid;
+    private final String appId;
+    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
     private final String id;
-    private final String sharedId;
-    
+    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
+    private final String originSharedId;
+
     @JsonCreator
     public AssessmentReference(@JsonProperty("guid") String guid,
-            @JsonProperty("id") String id, @JsonProperty("sharedId") String sharedId) {
-        this(INSTANCE, guid, id, sharedId);
+                               @JsonProperty("appId") String appId) {
+        this(INSTANCE, guid, null, null, appId);
     }
 
-    public AssessmentReference(ConfigResolver resolver, String guid, String id, String sharedId) {
+    public AssessmentReference(ConfigResolver resolver, String guid, String id, String originSharedId, String appId) {
```

Comment:
Usually I like to order parameters from broadest scope to narrowest scope, which means that appId almost always comes first in parameter lists, then usually followed by studyId and userId where they occur. I'm sure there's a lot of exceptions to this, but that's generally the pattern.